gitops/generate_build.go

```diff
@@ -105,8 +104,51 @@ func DetermineBuildExecution(component appstudiov1alpha1.Component, params []tek
 	return pipelineRunSpec
 }
 
-func determineBuildDefinition(component appstudiov1alpha1.Component) string {
-	return "devfile-build"
+// determineBuildPipeline should detect build pipeline to use for the component and return its name.
+// If it fails to autodetect right pipeline, noop pipeline will be returned.
+// If a repository consists of two parts (e.g. frontend and backend), it should be mapped to two components (see context field in CR).
+// Available build pipeleine are located here: https://github.com/redhat-appstudio/build-definitions/tree/main/pipelines
+func determineBuildPipeline(component appstudiov1alpha1.Component) string {
+	// It is possible to skip error checks here because the model is propogated by component controller
+	componentDevfileData, err := devfile.ParseDevfileModel(component.Status.Devfile)
+	if err != nil {
+		// Return dummy value for tests
+		return "noop"
+	}
+
+	// The only information about project is in language and projectType fileds under metadata of the devfile.
+	// They must be used to determine the right build pipeline.
+	devfileMetadata := componentDevfileData.GetMetadata()
+	language := devfileMetadata.Language
+	// TODO use projectType when build pipelines support frameworks
+	// projectType := devfileMetadata.ProjectType
+
+	switch strings.ToLower(language) {
+	case "java":
+		return "java-builder"
+	case "nodejs", "node":
+		return "nodejs-builder"
+	case "python":
+		// TODO return python-builder when the pipeline ready
+		return "noop"
+	}
+
+	// Failed to detect build pipeline
+	// Check for Dockerfile
+	filterOptions := devfilecommon.DevfileOptions{
+		ComponentOptions: devfilecommon.ComponentOptions{
+			ComponentType: devfilev1alpha2.ImageComponentType,
+		},
+	}
+	devfileComponents, _ := componentDevfileData.GetComponents(filterOptions)
+	for _, component := range devfileComponents {
+		if component.Image != nil && component.Image.Dockerfile != nil {
+			return "docker-build"
```

Comment:
Could we also get the path to the Dockerfile and pass that on to the Pipeline ? Without that, existing M2 workflows would break.

https://github.com/sbose78/devfile-sample-code-with-quarkus/blob/main/devfile.yaml#L27